sdk/clientcore/core/src/test/java/io/clientcore/core/http/models/PagedIterableTests.java

```diff
@@ -50,22 +50,23 @@ public String getNextLink() {
     }
 
     private PagedIterable<TodoItem> list() {
-        return new PagedIterable<>(() -> listSinglePage(), nextLink -> listNextSinglePage(nextLink));
+        return new PagedIterable<>((context) -> listSinglePage(null),
+            (context, nextLink) -> listNextSinglePage(null, nextLink));
     }
 
-    private PagedResponse<TodoItem> listSinglePage() {
-        Response<TodoPage> res = listSync();
+    private PagedResponse<TodoItem> listSinglePage(PagingOptions pagingOptions) {
+        Response<TodoPage> res = listSync(pagingOptions);
         return new PagedResponse<>(res.getRequest(), res.getStatusCode(), res.getHeaders(), res.getBody(),
             res.getValue().getItems(), res.getValue().getNextLink());
     }
 
-    private PagedResponse<TodoItem> listNextSinglePage(String nextLink) {
-        Response<TodoPage> res = listNextSync(nextLink);
+    private PagedResponse<TodoItem> listNextSinglePage(PagingOptions pagingOptions, String nextLink) {
+        Response<TodoPage> res = (nextLink == null) ? listSync(pagingOptions) : listNextSync(nextLink);
```

Comment:
Here, the code would fork in `nextLink`. If `nextLink` exists, it would just request that URL; if `nextLink` not exist, it would call the same API, but with `pagingOptions` updated (e.g. `continuationToken` updated from the prior page).